### PR TITLE
Expansion Set Logos

### PR DESCRIPTION
**Pull request by jwoolley**
Generates logos for new set announcements and uploads them as an image tweet with the headline.

There are a few caveats for this PR:

- PhantomJS must be installed and on the path
- GraphicsMagick must be installed and on the path
- The checked-in fonts must be installed
- Font textures (currently just `grit.png`) need to be uploaded to a web host
 (determined after many hours of trying to get it to work locally). 
    - The grammar that references it will need to be updated as well. Fortunately, if the the textures aren't accessible, the logos will work fine without them.
- Critically, we'll need to see if the fonts look decent enough when rendered on the VPS. :crossed_fingers: 

@mdiehr, the rendering should probably be tested before merging by running off of this branch. Alternatively, I can revert the relevant headline rules to not generate the logos and you can make the change locally to test it.

### CODE REVIEW

File: src/data/grammar/grammar.json [hunk 1]
```diff
@@ -32,13 +36,13 @@
         "PICTURES: #pro.capitalize#'s [_cosplayAdj:amazing,convincing,elaborate,unconventional]#_cosplayAdj# cosplay of [_who:#cardAlone#,#character#]#_who#",
         "New preview card from #teaserCardSet#! Is it a #descriptor# #realPowerfulCard#?"
     ],
-    "announcement": [
+    "announcement": [
```
Comment (pickfifteen) [inline src/data/grammar/grammar.json]: There's a few of these random whitespaces that seem incorrect throughout the diff, it'd be easier to read if they weren't here

File: src/data/grammar/grammar.json [hunk 2]
```diff
@@ -1,8 +1,5 @@
 {
     "origin": [
-        "#headline#"
```
Comment (pickfifteen) [inline src/data/grammar/grammar.json]: I disagree with this change; keeping the default origin as a one-item like `#headline#` allows the someone's config-override to confidently have a value that is the "normal" distribution even if that changes in the future. For instance the current config-override on the server is `#headline#, #headline#, #inGraphics#`. If we added a new top-level category to `#headline#`, the server would start using it without any config changes. Can't do that if all the top-level stuff is directly in `origin`, though.

File: src/data/grammar/grammar.json [hunk 3]
```diff
@@ -16,6 +13,13 @@
         "#inGraphics#",
         "#headlineKeepOrMulligan#"
     ],
+    "headline": [
+        "Wizards Announces: #novelCardSetName.capitalizeAll#",
+        "[_nextSet:The next set,The next fall set,#codename.capitalize#][_verbphrase:is actually,is really named,will be named]#_nextSet# #_verbphrase# #novelCardSetName.capitalizeAll#"
+    ],
+    "inGraphics": [
```
Comment (pickfifteen) [inline src/data/grammar/grammar.json]: I think in the latest, `inGraphics` is in `mtg-in-graphics` now so it shouldn't be in `grammar`? Confirm in your local copy post-merge from master, but I think this shouldn't be here now.

File: src/data/grammar/grammar.json (hunk 1 above)
Comment (mdiehr) [inline src/data/grammar/grammar.json]: You can append &w=1 at the end of the github url to ignore whitespace changes

File: src/data/grammar/grammar.json (hunk 3 above)
Comment (jwoolley) [inline src/data/grammar/grammar.json]: also inadvertent. I think this one was auto-merged this way.

File: src/data/grammar/grammar.json (hunk 2 above)
Comment (jwoolley) [inline src/data/grammar/grammar.json]: this happened when I resolved the merge, I guess I didn't understand the change as it appeared in the conflict block. I didn't revert it intentionally

File: bot.js [hunk 4]
```diff
@@ -52,6 +58,72 @@ function postSvgTweet(status, svgString, altText) {
...
+	return newHtml;
+}
+
+function renderImageFromHtml(html, outputPath) {
+	return new Promise((resolve, reject) => {
+		const screenshot = html2png({ width: 1024, height: 768, browser: 'phantomjs'});
+		
+		console.log('HTML:'); console.log(html);
+
+		screenshot.render(html, function (err, data) {
```
Comment (mdiehr) [inline bot.js]: much of this is duplicated in bot and index. possible to pull it out of both?

File: bot.js [hunk 5]
```diff
@@ -52,6 +58,72 @@ function postSvgTweet(status, svgString, altText) {
 		.catch(e => logError('Failed to post tweet: ' + e));
 }
 
+function resolveCssUrls(html) {
```
Comment (mdiehr) [inline bot.js]: Anything not directly related to the bot should be moved to another file (probably a new one)

File: bot.js (hunk 4 above)
Comment (pickfifteen) [inline bot.js]: Probably should be, along with the similar svg functionality

**Pull request merged**